packages/hardhat-chai-matchers/src/bigNumber.ts

```diff
@@ -0,0 +1,329 @@
+// TODO: consider adding support for the following Chai.Assert properties.
+// Might any of these come "for free" due to supporting other methods? At the
+// very least get them tested.
+//   isAbove
+//   isAtLeast
+//   isBelow
+//   isAtMost
+//   isAt
+//   isNumber
+//   isNotNumber
+//   closeTo
+//   approximately
+
+import type { BigNumber as EthersBigNumberType } from "ethers";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import type { BigNumber as BigNumberJsType } from "bignumber.js";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import type { default as BNType } from "bn.js";
+
+function isBN(n: any) {
+  try {
+    // eslint-disable-next-line import/no-extraneous-dependencies
+    const BN: typeof BNType = require("bn.js");
+    return BN.isBN(n);
+  } catch (e) {
+    return false;
+  }
+}
+
+function isEthersBigNumber(n: any) {
+  try {
+    // eslint-disable-next-line import/no-extraneous-dependencies
+    const BigNumber: typeof EthersBigNumberType =
+      require("ethers").ethers.BigNumber;
+    return BigNumber.isBigNumber(n);
+  } catch (e) {
+    return false;
+  }
+}
+
+function isBigNumberJsBigNumber(n: any) {
+  try {
+    // eslint-disable-next-line import/no-extraneous-dependencies
+    const BigNumber: typeof BigNumberJsType = require("bignumber.js").BigNumber;
+    return n instanceof BigNumber && BigNumber.isBigNumber(n);
```

Comment:
I think we discussed this but I don't remember right now. Why do we use `instanceof` here but not in the other checks?

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/2428/commits/f701a2ba9ff1db93039535bf51b2188903f938a4. I had removed the others previously but I missed this one. Good catch!